integration-test/test/test_helper.bash

```diff
@@ -0,0 +1,126 @@
+# Test helpers for BATS tests
+
+IMAGE=custom-script
+DOCKERFILE=test.Dockerfile
+TEST_CONTAINER=test
+
+certs_dir="$BATS_TEST_DIRNAME/certs"
+
+build_docker_image() {
+    echo "Building test image..."
+    docker build -q -f $DOCKERFILE -t $IMAGE . 1>&2
+}
+
+in_tmp_container() {
+    docker run --rm $IMAGE "$@"
+}
+
+rm_container() {
+    docker rm -f $TEST_CONTAINER &>/dev/null && \
+        echo "Deleted test container." || true
+}
+
+mk_container() {
+    rm_container && echo "Creating test container with commands: $@">&2 && \
+        docker create --name=$TEST_CONTAINER $IMAGE "$@" 1>/dev/null
+}
+
+in_container() {
+    set -e
+    rm_container
+    mk_container "$@"
+    echo "Starting test container...">&2
+    start_container
+}
+
+start_container() {
+    docker start --attach $TEST_CONTAINER
+}
+
+container_diff() {
+    docker diff $TEST_CONTAINER
+}
+
+container_read_file() { # reads the file at container path $1
+    set -eo pipefail
+    docker cp $TEST_CONTAINER:"$1" - | tar x --to-stdout
+} 
+
+mk_certs() { # creates certs/{THUMBPRINT}.(crt|key) files under ./certs/ and prints THUMBPRINT
+    set -eo pipefail
+    mkdir -p "$certs_dir" && cd "$certs_dir" && rm -f "$certs_dir/*"
+    openssl req -x509 -newkey rsa:2048 -keyout key.pem -out cert.pem -days 365 -nodes -batch &>/dev/null
+    thumbprint=$(openssl x509 -in cert.pem -fingerprint -noout| sed 's/.*=//g' | sed 's/://g')
```

Comment:
😁 The variable is called thumbprint, but the command is fingerprint.  You've been here too long!